explainer.md

```diff
@@ -47,66 +47,46 @@ On a system that does not use file extensions but associates files with MIME typ
+This would create a new top level browsing context, navigating to '{origin}{action}?name={HANDLER_NAME}'. Assuming the user opened `graph.csv` in Graphr the url would be `https://graphr.com/open-files/?name=raw`. When the `load` event is fired, an additional `launchParams` property will be available on global object.
 
-> Note: `load` is possibly not the correct place for this. We are considering other options, including `DOMContentLoaded` or a completely new event.
+> Note: The `launchParams` property is discussed in more detail in the [Launch Events](https://github.com/WICG/sw-launch) explainer.
 
-The shape of `LoadEvent` and `LaunchParams` is described below:
+The shape of `LaunchParams` is described below:
 ```cs
 interface LaunchParams {
-  // Cause of the launch (e.g. file_handler|share_target|shortcut|link). Initially only file_handler will be supported but more will likely be added in future.
-  // The values of this enum should be based on entries in the manifest (such as 'file_handler' and 'share_target'), where appropriate.
-  readonly attribute DOMString cause;
-  // The files the application was launched with. 
-  sequence<FileSystemFileHandle>? files;
-}
+  // The files (if any) the application was launched with.
+  sequence<FileSystemFileHandle>? fileHandles;
 
-interface LoadEvent : Event {
-  // An instance of the LaunchParams object above, detailing how the launch happened.
-  attribute LaunchParams launchParams;
+  // The request which is the cause of the launch. This is a copy of the request that is sent to the server.
+  Request request;
 }
 ```
 
-An application could then extract the [FileSystemFileHandles](https://github.com/WICG/native-file-system/blob/master/EXPLAINER.md) in a `load` event listener, and handle the files.
+An application could then choose how to handle the files it was launched with. We recommend saving the file handle to disk and creating a url to address the launched file. This way a user can navigate back to a file they had open.
 
 ```js
+// In graphr.com/open-files
 window.addEventListener('load', event => {
   // Launch params could be undefined if the browser doesn't support it.
-  if (!event.launchParams || !event.launchParams.cause === 'file_handler')
-    return;
-
-  const fileHandles = event.launchParams.files;
-  // TODO: Handle the files.
-});
-```
-
-This API is sufficient to allow native files to be opened in web applications. However, it doesn't cater for more advanced use cases, such as opening a file in an existing window, or simply displaying a notification when a file is opened. For these cases applications can add a [launch event handler](https://github.com/WICG/sw-launch/blob/master/explainer.md). The `launchParams` object would be attached to the `launch` event in an identical manner to the `load` event. 
-
-```js
-self.addEventListener('launch', event => {
-  if (event.launchParams.cause !== 'files')
+  if (!window.launchParams || window.launchParams.cause !== 'file_handler')
```

Comment:
nit: cause is gone now. Should we do something else instead?

Reply:
Whoops good catch!